cfme/tests/ansible/test_embedded_ansible_automate.py

```diff
@@ -125,30 +128,28 @@ def ansible_catalog_item(ansible_repository):
         cat_item.delete()
 
 
-@pytest.yield_fixture
-def custom_vm_button(appliance, ansible_catalog_item):
+@pytest.yield_fixture(params=["Localhost", "Target Machine", "Specific Hosts"],
+                      ids=["localhost", "target_machine", "specific_hosts"])
+def custom_vm_button(appliance, ansible_catalog_item, request, vmware_vm):
     buttongroup = appliance.collections.button_groups.create(
         text=fauxfactory.gen_alphanumeric(),
         hover="btn_desc_{}".format(fauxfactory.gen_alphanumeric()),
         type=appliance.collections.button_groups.VM_INSTANCE)
+    hosts = vmware_vm.ip_address if request.param == "Specific Hosts" else None
     button = buttongroup.buttons.create(
+        type="Ansible Playbook",
         text=fauxfactory.gen_alphanumeric(),
         hover="btn_hvr_{}".format(fauxfactory.gen_alphanumeric()),
-        dialog=ansible_catalog_item.provisioning["provisioning_dialog_name"],
```

Comment:
Question: We don't need dialog here anymore?

Reply:
Correct, a new type of buttons were introduced in CFME 5.9 - Ansible Playbook. It makes the creation of buttons which execute ansible playbooks easier.